### PR TITLE
[Merged by Bors] - feat(Analysis/Distribution/SchwartzMap): projection to ZeroAtInfty and corresponding type class

### PR DESCRIPTION
**Pull request by mcdoll**
Adds a characterization of ZeroAtInfty in terms of norms and uses one direction to show that every Schwartz function is zero at infinity.
We also add a few lemmas that characterize elements of the `cocompact` filter in terms of norm estimates.

---

Part of a project to prove integration by parts for the partial derivatives on Schwartz space.

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Topology/MetricSpace/Bounded.lean [hunk 1]
```diff
@@ -172,6 +172,17 @@ theorem cobounded_le_cocompact : cobounded α ≤ cocompact α :=
 #align comap_dist_right_at_top_le_cocompact Metric.cobounded_le_cocompactₓ
 #align comap_dist_left_at_top_le_cocompact Metric.cobounded_le_cocompactₓ
 
+theorem _root_.Bornology.IsCobounded.closedBall_compl_subset {s : Set α} (hs : IsCobounded s)
```
Comment (sgouezel) [inline Mathlib/Topology/MetricSpace/Bounded.lean:175]: Could you turn this into an iff (and register the current implication as an alias)?

File: Mathlib/Topology/MetricSpace/ProperSpace.lean [hunk 2]
```diff
@@ -153,6 +153,12 @@ theorem exists_lt_subset_ball (hs : IsClosed s) (h : s ⊆ ball x r) : ∃ r' <
   · exact (exists_pos_lt_subset_ball hr hs h).imp fun r' hr' => ⟨hr'.1.2, hr'.2⟩
 #align exists_lt_subset_ball exists_lt_subset_ball
 
+theorem mem_cocompact_of_exists_closedBall_compl_subset (x : α) (h : ∃ r, (closedBall x r)ᶜ ⊆ s) :
```
Comment (sgouezel) [inline Mathlib/Topology/MetricSpace/ProperSpace.lean:156]: ditto

**Review by sgouezel — COMMENTED** [review]
bors d+

**Comment by mcdoll** [comment]
bors merge

**Pull request closed without merging**